plugin-server/src/utils/db/sql.ts

```diff
@@ -25,11 +25,38 @@ function pluginConfigsInForceQuery(specificField?: keyof PluginConfig): string {
 
 export async function getPluginRows(hub: Hub): Promise<Plugin[]> {
     const { rows }: { rows: Plugin[] } = await hub.db.postgresQuery(
+        // `posthog_plugin` columns have to be listed individually, as we want to exclude the blob `archive` column,
+        // and Postgres unfortunately doesn't have a feature to exclude a single column from *
         `SELECT
```

Comment:
Q: This query took ~13 seconds before individually and we're adding to this with a new join here - not sure how this is resolving the performance problem outlined in https://github.com/PostHog/posthog/issues/10308 which was focussed on the query only.

Reply:
The query itself, as in SQL, is not an issue. It's reasonably well optimized with regard to lookups (incl. joins) and there's just not many rows. What scales badly is the `archive` column. It takes time to transfer all this and unpack it in the plugin server each time. Locally it's 10-30 s with the `archive` column and 0.3-0.5 s when excluding it. In the production Django shell it's 2-3 s vs. 0.3-0.5 s. (This does not include unpacking.)

TBH only now did I notice that there's one big offender which is the main source of our current woes – turns out that https://github.com/PostHog/posthog-variance-plugin includes the `.yarn` folder, which is 40 MB in size. All the other archives are 10 MB _in total_, so optimizing the Variance plugin could go a long way.

However, I think it's still worth it to go through this refactor, as with the optimized scheme this PR introduces we will hopefully not have to worry about archive sizes in the plugin server for a long time. We've now also got metrics to track this.

Reply:
It is the most time-consuming query we do on postgres (total time spent): https://metabase.posthog.net/question/344-most-time-consuming-postgres-queries. This means that during deploys we risk bringing down capture and decide endpoints and causing other load-related issues.

Can you run and compare the new and old queries - will we see a performance win from removing the `archive` column?

> However, I think it's still worth it to go through this refactor, as with the optimized scheme this PR introduces we will hopefully not have to worry about archive sizes in the plugin server for a long time. We've now also got metrics to track this.

Not arguing that - I think this is great, but orthogonal to the goal in https://github.com/PostHog/posthog/issues/10308 if it doesn't reduce the time spent in postgresql.

Reply:
I think I won't get the metrics I'm looking for here. Updating the PR description to remove the "closes" note as we'll need to monitor and IMO likely need to make file loading lazy in the future.

Reply:
Very well, I agree it's not a slam dunk, let's monitor.